tripal_analysis_expression/includes/TripalFields/local__expression_search/local__expression_search_formatter.inc

```diff
@@ -48,61 +45,85 @@ class local__expression_search_formatter extends ChadoFieldFormatter
     ])->cvterm_id;
     $analysis_id = $entity->chado_record->analysis_id;
 
-    $query = chado_db_select('analysis', 'a');
-    $query->distinct();
-    $query->fields('f', ['organism_id']); //feature name and id for
-    $query->fields('fc', ['cvterm_id']);
-    $query->fields('ct', ['name']); // Keyword
-    $query->join('analysisfeature', 'af', 'a.analysis_id = af.analysis_id');
-    $query->join('feature', 'f', 'f.feature_id = af.feature_id');
-    $query->join('feature_cvterm', 'fc', 'f.feature_id = fc.feature_id');
-    $query->join('feature_cvtermprop', 'fp', 'fc.feature_cvterm_id = fp.feature_cvterm_id AND fp.type_id = :typeid', [':typeid' => $pvalue_cvterm_id]);
-    $query->join('feature_cvtermprop', 'fp2', 'fp.feature_cvterm_id = fp2.feature_cvterm_id AND fp2.type_id = :typeid2', [':typeid2' => $evidence_code_cvterm_id]);
-    $query->join('cvterm', 'ct', 'fc.cvterm_id = ct.cvterm_id');
-    $query->condition('a.analysis_id', $analysis_id, '='); // commented out for testing so I have multiple CVterms
-    $total = $query->countQuery()->execute()->fetchField();
-    $page = pager_default_initialize($total, 10);
-    $query->range($page * 10, 10);
-    $cvterms = $query->execute()->fetchAll();
+    $sql = "
+    SELECT ct.cvterm_id, ct.name, COUNT(f.feature_id) as count FROM chado.analysis a
+    INNER JOIN chado.analysisfeature af ON a.analysis_id = af.analysis_id
+    INNER JOIN chado.feature f ON f.feature_id = af.feature_id
+    INNER JOIN chado.feature_cvterm fc ON f.feature_id = fc.feature_id
+    INNER JOIN chado.feature_cvtermprop fp ON fc.feature_cvterm_id = fp.feature_cvterm_id AND fp.type_id = :typeid
+    INNER JOIN chado.feature_cvtermprop fp2 ON fp.feature_cvterm_id = fp2.feature_cvterm_id AND fp2.type_id = :typeid2
+    INNER JOIN chado.cvterm ct ON fc.cvterm_id = ct.cvterm_id
+    WHERE a.analysis_id = :analysis_id
+    GROUP BY ct.cvterm_id
+    ";
+
+    $cvterms = db_query($sql, [
+      ':typeid' => $pvalue_cvterm_id,
+      ':typeid2' => $evidence_code_cvterm_id,
+      ':analysis_id' => $analysis_id,
+    ])->fetchAll();
+
+    $organism_query = "SELECT * FROM chado.analysis_organism WHERE analysis_id=:id";
+    $organism = db_query($organism_query, [':id' => $analysis_id])->fetchObject();
```

Comment:
Another thing here is that we are getting the organism information from a different table in a much simpler query. Doing this makes it faster for the previous query do the counting and not having to find the organism for each cvterm.